chromadb/utils/environment.py

```diff
@@ -0,0 +1,19 @@
+import sys
+
+
+# https://stackoverflow.com/a/39662359
+def running_inside_notebook() -> bool:
+    try:
+        shell = get_ipython().__class__.__name__  # type: ignore
+        if shell == "ZMQInteractiveShell":
+            return True  # Jupyter notebook or qtconsole
+        elif shell == "TerminalInteractiveShell":
+            return False  # Terminal running IPython
+        else:
+            return False  # Other type (?)
+    except NameError:
+        return False  # Probably standard Python interpreter
+
+
+def running_in_interactive_environment() -> bool:
```

Comment:
do we also want to add this check if we're using the batch upsert from utils?

Reply:
by this do you mean automatically displaying progress when people directly use the batch util as well? that seems ok